### PR TITLE
`Development`: Improve client test coverage for assessment components

### PR DESCRIPTION
**Pull request by imstreetspirit**
#### General

- [ ]* I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ]* This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server (not needed)
- [ ]* I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [ ]* I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ]* I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [ ]* I implemented the changes with a good performance and prevented too many database calls.
- [ ]* I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [ ]* I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ]* I documented the TypeScript code using JSDoc style.
- [ ]* I added multiple screenshots/screencasts of my UI changes.
- [ ]* I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises (not needed)
- [ ]* I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ]* I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

Increased client test coverage for several test units

#### Code Review
- [x] Review 1
- [x] Review 2
### Test Coverage

example-text-submission.component.ts - 86% (former 77%)
unreferenced-feedback.component.ts - 95% (former 12%)
assessment-instructions.component.ts - 100% (former 23%)
structured-grading-instructions-assessment-layout.component.ts - 95% (former 29%)

### CODE REVIEW

**Review by willergeorg — APPROVED** [review]
Newly added test cases look good to me. The comments I added are regarding old test code that is not following our client test code guidelines. It would be nice if you would change it but I understand that this is not necessarily in scope of your PR.

**Comment by imstreetspirit** [comment]
> Newly added test cases look good to me. The comments I added are regarding old test code that is not following our client test code guidelines. It would be nice if you would change it but I understand that this is not necessarily in scope of your PR.

I can fix them easily, thanks for noticing...

File: src/test/javascript/spec/component/text-exercise/example-text-submission.component.spec.ts [hunk 1]
```diff
@@ -238,12 +238,12 @@ describe('ExampleTextSubmissionComponent', () => {
...
-        expect(comp.submission?.results).toBeUndefined();
-        expect(comp.submission?.latestResult).toBeUndefined();
-        expect(comp.result).toBeUndefined();
-        expect(comp.textBlockRefs).toHaveLength(0);
-        expect(comp.unusedTextBlockRefs).toHaveLength(0);
+        expect(comp.submission?.blocks).toBe(undefined);
+        expect(comp.submission?.results).toBe(undefined);
+        expect(comp.submission?.latestResult).toBe(undefined);
+        expect(comp.result).toBe(undefined);
+        expect(comp.textBlockRefs.length).toBe(0);
```
Comment (willergeorg) [inline src/test/javascript/spec/component/text-exercise/example-text-submission.component.spec.ts:245]: Oh no what do you have against `toHaveLength()`? Can we undo these changes again? 😅

**Review by willergeorg — COMMENTED** [review]
Thanks for incorporating my feedback! Just one small misunderstanding left...

**Review by willergeorg — APPROVED** [review]

**Review by cremertim — APPROVED** [review]
Great work increasing the coverage, code lgtm 👍
